direct/nn/iterdualnet/iterdualnet.py

```diff
@@ -0,0 +1,201 @@
+    in the image and k-space domain.
+    """
+
+    def __init__(
+        self,
+        forward_operator: Callable,
+        backward_operator: Callable,
+        num_iter: int = 10,
+        image_normunet: bool = False,
+        kspace_normunet: bool = False,
+        image_no_parameter_sharing: bool = True,
+        kspace_no_parameter_sharing: bool = True,
+        compute_per_coil: bool = True,
+        **kwargs,
+    ):
+        """Inits :class:`IterDualNet`.
+
+        Parameters
+        ----------
+        forward_operator : Callable
+            Forward Operator.
+        backward_operator : Callable
+            Backward Operator.
+        num_iter : int
+            Number of iterations. Default: 10.
+        image_normunet : bool
+            If True will use NormUNet for the image model. Default: False.
+        kspace_normunet : bool
+            If True will use NormUNet for the kspace model. Default: False.
+        image_no_parameter_sharing : bool
+            If False, a single image model will be shared across all iterations. Default: True.
+        kspace_no_parameter_sharing : bool
+            If False, a single kspace model will be shared across all iterations. Default: True.
+        compute_per_coil : bool
+            If True :math:`f` will be transformed into a multi-coil kspace.
+        kwargs : dict
+            Kwargs for unet models.
+        """
+        super().__init__()
+
+        self.forward_operator = forward_operator
+        self.backward_operator = backward_operator
+        self.num_iter = num_iter
+
+        self.image_no_parameter_sharing = image_no_parameter_sharing
+        self.kspace_no_parameter_sharing = kspace_no_parameter_sharing
+        image_unet_architecture = NormUnetModel2d if image_normunet else UnetModel2d
+        kspace_unet_architecture = NormUnetModel2d if kspace_normunet else UnetModel2d
+
+        self.image_block_list = nn.ModuleList()
+        self.kspace_block_list = nn.ModuleList()
+
+        for _ in range(self.num_iter if self.image_no_parameter_sharing else 1):
+            self.image_block_list.append(
+                image_unet_architecture(
+                    in_channels=2,
+                    out_channels=2,
+                    num_filters=kwargs.get("image_unet_num_filters", 8),
+                    num_pool_layers=kwargs.get("image_unet_num_pool_layers", 4),
+                    dropout_probability=kwargs.get("image_unet_dropout", 0.0),
+                )
+            )
+        for _ in range(self.num_iter if self.kspace_no_parameter_sharing else 1):
+            self.kspace_block_list.append(
+                kspace_unet_architecture(
+                    in_channels=2,
+                    out_channels=2,
+                    num_filters=kwargs.get("kspace_unet_num_filters", 8),
+                    num_pool_layers=kwargs.get("kspace_unet_num_pool_layers", 4),
+                    dropout_probability=kwargs.get("kspace_unet_dropout", 0.0),
```

Comment:
Same concern here.

Reply:
2 is the number of channels (real, imag) and is used across all DIRECT for complex tensors so for anyone familiar with the code can understand that this is not a magic number. But thanks for the comment.